src/js/tech/ChromecastTech.js

```diff
@@ -0,0 +1,364 @@
+      this._currentSource = options.source;
+      this._requestTitle = options.requestTitleFn || _.noop;
+      this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      // See `currentTime` function
+      this._firstItemStartTime = options.startTime || 0;
+      // See `duration` function
+      this._firstItemDuration = options.source.duration || 0;
+
+      this._playSource(options.source, this._firstItemStartTime);
+      this.ready(function() {
+         this.setMuted(options.muted);
+      }.bind(this));
+
+      return subclass;
+   },
+
+   createEl: function() {
+      return this._ui.getDOMElement();
+   },
+
+   play: function() {
+      if (!this.paused()) {
+         return;
+      }
+      if (this.ended() && !this._isMediaLoading) {
+         // Restart the current item from the beginning
+         this._playSource(this._currentSource, 0);
+      } else {
+         this._remotePlayerController.playOrPause();
+      }
+   },
+
+   pause: function() {
+      if (!this.paused() && this._remotePlayer.canPause) {
+         this._remotePlayerController.playOrPause();
+      }
+   },
+
+   paused: function() {
+      return this._remotePlayer.isPaused || this.ended();
+   },
+
+   setSource: function(source) {
+      // Store the current source so that we have a reference to it if we need to restart
+      // the current item.
+      this._currentSource = source;
+      this._playSource(source, 0);
+   },
+
+   _playSource: function(source, startTime) {
+      var castSession = this._getCastSession(),
+          mediaInfo = new chrome.cast.media.MediaInfo(source.src, source.type),
+          title = this._requestTitle(source),
+          subtitle = this._requestSubtitle(source),
+          request;
+
+      this.trigger('waiting');
+      this._clearSessionTimeout();
+
+      mediaInfo.metadata = new chrome.cast.media.GenericMediaMetadata();
+      mediaInfo.metadata.metadataType = chrome.cast.media.MetadataType.GENERIC;
+      mediaInfo.metadata.title = title;
+      mediaInfo.metadata.subtitle = subtitle;
+
+      this._ui.updateTitle(title);
+      this._ui.updateSubtitle(subtitle);
+
+      request = new chrome.cast.media.LoadRequest(mediaInfo);
+      request.autoplay = true;
+      request.currentTime = startTime;
+
+      this._isMediaLoading = true;
+      castSession.loadMedia(request)
+         .then(function() {
+            if (!this._hasPlayedAnyItem) {
+               // `triggerReady` is required here to notify the Video.js player that the
+               // Tech has been initialized and is ready.
+               this.triggerReady();
+            }
+            this.trigger('loadstart');
+            this.trigger('play');
+            this.trigger('playing');
+            this._hasPlayedAnyItem = true;
+            this._isMediaLoading = false;
+         }.bind(this), this._triggerErrorEvent.bind(this));
+   },
+
+   setCurrentTime: function(time) {
+      var currentDuration = this.duration();
```

Comment:
Why is the duration "current"? It doesn't seem anything in this function would cause the duration to change. I would think `duration` is sufficient.